src/mnemonic.rs

```diff
@@ -136,7 +137,11 @@ impl Mnemonic {
     where
         S: Into<String>,
     {
-        let phrase = phrase.into();
+        let phrase: String = phrase
+            .into()
+            .split(" ")
+            .filter(|word| !word.is_empty())
```

Comment:
You could do `phrase.into().split_whitespace().join(" ")`, that will handle all sorts of whitespace. Also, since this is creating a new `String`, there is no reason to convert phrase, so `S: AsRef<str>` instead of `S: Into<String>` would be more sensible. Removing the generic in favor of a regular `phrase: &str` would be fine too.